test/utils/test_serialization.py

```diff
@@ -10,8 +10,8 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-from os import remove
-from tempfile import mkstemp
+import os
+from tempfile import TemporaryDirectory
```

Comment:
Awesome! I did not know about this, why not use `TemporaryFile` instead?

Reply:
`TemporaryFile` is like `mkstemp` in that it gives you a file handle.
```python
>>> with tempfile.TemporaryFile() as fp:
...     fp.write(b'Hello world!')
...     fp.seek(0)
...     fp.read()
```